### PR TITLE
PF-716 Add private controlled AI Notebook instance client tests

### PR DESCRIPTION
**Pull request by wchamber**
Adds a private controlled AI Notebook instance lifecycle test.
One outstanding todo on checking permissions given our project level service account permissions. Otherwise I'm not trying to write any other client tests than what's in this PR, so if more are desired, please let me know.

### CODE REVIEW

File: integration/src/main/java/scripts/testscripts/PrivateControlledAiNotebookInstanceLifecycle.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+  @Override
+  protected void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+      throws Exception {
+    super.doSetup(testUsers, workspaceApi);
+    // Note the 0th user is the owner of the workspace, set up in the super class and passed as the
+    // 'testUser'.
+    assertThat(
+        "There must be at least three test users defined for this test.",
+        testUsers != null && testUsers.size() > 1);
```
Comment (zloery) [inline integration/src/main/java/scripts/testscripts/PrivateControlledAiNotebookInstanceLifecycle.java:67]: `testUsers.size() > 2`

**Review by zloery — APPROVED** [review]
LGTM

File: integration/src/main/java/scripts/testscripts/PrivateControlledAiNotebookInstanceLifecycle.java (hunk 1 above)
Comment (wchamber) [reply integration/src/main/java/scripts/testscripts/PrivateControlledAiNotebookInstanceLifecycle.java:67]: done

**Review by ddietterich — APPROVED** [review]
lgtm

**Pull request merged**